### PR TITLE
upgrade to rocksdb 0.18. enable jemalloc

### PR DESCRIPTION
**Pull request by Voxelot**
Upgrade rocksdb, enable jemalloc.

Also, change compression algorithm to lz4 as that is recommended by RocksDB docs. Disable inclusion of alternate unused compression schemes.

### CODE REVIEW

**Comment by rakita** [comment]
Can we merge https://github.com/FuelLabs/fuel-core/pull/381 first and then enable jemalloc to not introduce merge conflicts?

**Review by adlerjohn — COMMENTED** [review]
looks like CI is failing

File: fuel-core/Cargo.toml [hunk 1]
```diff
@@ -53,9 +53,13 @@ itertools = "0.10"
 lazy_static = "1.4"
 prometheus = { version = "0.13", optional = true }
 rand = "0.8"
-rocksdb = { version = "0.17", features = [
+rocksdb = { version = "0.18", default-features = false, features = [
+    "zlib",
+    "zstd",
+    "lz4",
```
Comment (adlerjohn) [inline fuel-core/Cargo.toml:59]: What are the ramifications of this?

**Review by rakita — APPROVED** [review]
lgtm

**Pull request merged**